types/state_aware_marker.go

```diff
@@ -0,0 +1,154 @@
+//go:generate mockgen -package types -destination appender_mock_test.go . StateAppender
+package types
+
+import (
+	"slices"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+
+	_ "github.com/marcboeker/go-duckdb"
+)
+
+type marker interface {
+	AlertMarker
+	GroupMarker
+}
+
+// FushableMarker is a marker that you should flush before you attempt to read state data
+type FushableMarker interface {
+	marker
+	Flush() error
+}
+
+type StateAppender interface {
+	Append(fingerprint model.Fingerprint, state AlertState)
+	AppendInhibited(fingerprint model.Fingerprint, inhibitedBy []string)
+	Close() error
+	Flush() error
+}
+
+func NewStateAwareMarker(
+	r prometheus.Registerer,
+	appender StateAppender,
+) *StateAwareMarker {
+	return &StateAwareMarker{
+		marker:   NewMarker(r),
+		appender: appender,
+	}
+}
+
+type StateAwareMarker struct {
+	marker   marker
+	appender StateAppender
+}
+
+func (s *StateAwareMarker) SetActiveOrSilenced(
```

Comment:
What is this suppose to do? `SetActiveOrSilenced` sounds like it's updating alert state, but in the it's conditionally appending it somewhere. Maybe some comments would help?

Reply:
This is also interface implementation, I could add some comments on the Struct itself indicating that this is the same implementation as the original one but just storing the final state to a db.

But these methods are documented on the interface 👀 
https://github.com/prometheus/alertmanager/blame/c88eb78142c80a8046204650d02ffc7a71cdceda/types/types.go#L73